### ISSUE TITLE
Flexilayouts follow up: Region machine names should be unique

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#4045]
This is a follow-up to https://github.com/backdrop/backdrop-issues/issues/4024

When trying to build a select list of available regions, it appears that their machine names are non-unique. For example, the second region in row 1 has a machine name of `region_name_1` as does the second region in row 2.  We might want consider cleaning these up to make it easier to work with all regions at once.

```
<select>
  <option name="region_name_1">Region 1: Row 1</option>
  <option name="region_name_1">Region 1: Row 2</option>
  <option name="region_name_2">Region 2: Row 1</option>
  <option name="region_name_2">Region 2: Row 2</option>
</select>
```

**Comment by klonos** [comment]
This also affects `/admin/reports/blocks`: #4072

**Comment by jenlampton** [comment]
I don't think it's the same thing, as in the Blocks UI we can switch to using the Human-readable names. This issue was more about what happens when you have them all in the same select list and need both the Human name and the machine name.  I'll add a code sample above.

**Comment by klonos** [comment]
> I don't think it's the same thing...

Yup, that's why I moved it into it's own, separate issue (and filed a PR for it 😉). The root cause is kinda the same though: the regions (`positions`) returned by `layout_load_multiple()` are UUIDs, instead of the human-readable names.
